dvc/output/base.py

```diff
@@ -135,6 +136,9 @@ def __init__(
         self.path_info = self._parse_path(tree, path)
         if self.use_cache and self.cache is None:
             raise RemoteCacheRequiredError(self.path_info)
+        self.is_user_executable = os.path.isfile(path) and os.access(
+            path, os.X_OK
+        )
```

Comment:
The file might not exist here or might not be in the same state as the rest of the info. In this place we should load the isexec from the info provided, e.g. how `metric` is loaded above.

The actual stat should happen in `save()` method, as pointed out in https://github.com/iterative/dvc/pull/5036#discussion_r537530670 .

Reply:
Thank you for the clarification. I moved `stat` to `save()`.